### PR TITLE
Add option for extra certificate authorities.

### PR DESCRIPTION
**Pull request by calavera**
This allows using custom certificates without having to install their CAs in the system pool, and without having to skip their verification.

The configuration is something like this:

```
export let options = {
  tlsExtraCAs: [
      open(./ca1.pem),
      open(./ca2.pem)
  ]
}
```

I'm not completely sure how to add tests for this, tbh. Help very welcome

Signed-off-by: David Calavera <david.calavera@gmail.com>

### CODE REVIEW

File: lib/options.go [hunk 1]
```diff
@@ -327,6 +327,7 @@ type Options struct {
 	TLSCipherSuites *TLSCipherSuites `json:"tlsCipherSuites" envconfig:"K6_TLS_CIPHER_SUITES"`
 	TLSVersion      *TLSVersions     `json:"tlsVersion" ignored:"true"`
 	TLSAuth         []*TLSAuth       `json:"tlsAuth" envconfig:"K6_TLSAUTH"`
+	TLSExtraCAs     []string         `json:"tlsExtraCAs" envconfig:"K6_TLS_EXTRA_CAS"`
```
Comment (MStoykov) [inline lib/options.go:330]: You aren't merging the settings between the `json` and the `env`, please look at how it's done for other configurations and copy-paste them
You also don't support this to be configurable from a cli option, although I don't think that is required

File: js/runner.go [hunk 2]
```diff
@@ -158,6 +159,21 @@ func (r *Runner) newVU(id int64, samplesOut chan<- stats.SampleContainer) (*VU,
 		}
 	}
 
+	pool, _ := x509.SystemCertPool()
+	if pool == nil {
+		// SystemCertPool is not available on Windows.
+		// x509.SystemCertPool will always fail in that environment.
+		// The error can be ignored, instead, we initialize a new pool.
+		pool = x509.NewCertPool()
```
Comment (MStoykov) [inline js/runner.go:167]: That will probably need testing ... :sob:

**Review by MStoykov — COMMENTED** [review]
Hi, we just released v0.28.0 (as you've probably noticed), and will have some cleanup tasks, so we might not be .. very fast in answering ;(/

I have left a couple of comments ... additionally I would have to add that the ~biggest blocker (IMO) is that this won't work ... with `k6 cloud` or `k6 archive` all that well and that seems kind of important to me.~ None of this is true as you are using `open` so it will be part of the archive so that isn't a blocker 

~When I have time I will try to look into a possible solution, but you can work on that if you want as well. The idea is that `k6 archive` (unless bugs) will make an `archive.tar` that you can move to another machine and `k6 run archive.tar` and it will .. work :tm: . So for example in this case that means that we need to actually put the CA roots in the archive and then load them from there ... which might turn out to be harder than expected as I don't think we have a similar problem for another option.
`k6 cloud` is basically `k6 archive`++~

File: lib/options.go (hunk 1 above)
Comment (calavera) [reply lib/options.go:330]: good call, I have not looked at that code before, and I need to do it

File: js/runner.go (hunk 2 above)
Comment (calavera) [reply js/runner.go]: yeah, I've done it in the past, but I'll have to check how tests bootstrap data in this project to do that.

**Comment by MStoykov** [comment]
Hi @calavera , sorry for the long radio silence, are you still willing to work on this, if not we can probably pick it up?

It seems I was *very* wrong in my previous review, and the main blocker(not working with k6 cloud/archive) is actually not a blocker as you are using `open` and that already does what is required :tada: and then just using the contents of the `pem` files instead of their names on disk in the configuration.

This whole PR definitely needs at least 1 unit test though

Also because of the way `options` is saved as JSON in the archive, the whole `pem` file will be inside of it. This also makes it kind of hard to use the environmental variable as you will need to have the value be the contents of the `pem` file, not its name(having the name will require support for the archive/cloud).

I am now wondering if instead of this being in `options` we shouldn't add a new module "k6/tls" and have a way to configure it from there as in(names are illustrative only):

```javascript
import tls from "k6/tls"
tls.addExtraCA(open("./ca.pem"));
```
#1385 might be better off being in there as well.
Although at that point this might be too involved and actually should be left for when we design the new HTTP API? Arguably this could be how you configure it globally and when we are designing the API we will have a way to get a `new tls.Config()` that can be configured as well. 

cc @na-- @imiric

**Comment by calavera** [comment]
> are you still willing to work on this, if not we can probably pick it up?

please, feel free to take over these changes. I'm too busy with other things to be a great driver for them. Don't hesitate to close this PR if you think this would be better implemented in a different way.

**Pull request closed without merging**